src/Model/Import/MediaObject/MediaManagerImageCollectionFactory.php

```diff
@@ -46,20 +45,20 @@ function (MediaObjectReference $mediaObjectReference) {
                     return null;
                 }
 
-                return new Image(
-                    $mediaObjectAggregate->getMediaObjectId(),
+                return new MediaImage(
+                    $id,
                     $mediaObjectAggregate->getMimeType(),
-                    new Description($mediaObjectReference->getDescription()->toString()),
-                    $mediaObjectReference->getCopyrightHolder(),
+                    new Description($image->getDescription()->toString()),
+                    $image->getCopyrightHolder(),
                     $mediaObjectAggregate->getSourceLocation(),
-                    new Language($mediaObjectReference->getLanguage()->toString())
+                    $image->getLanguage()
                 );
             },
-            $mediaObjectReferences->toArray()
+            $images->toArray()
         );
 
-        $images = array_filter($images);
+        $imageCollection = array_filter($imageCollection);
 
-        return ImageCollection::fromArray($images);
+        return ImageCollection::fromArray($imageCollection);
```

Comment:
```suggestion
        return ImageCollection::fromArray($mediaImages);
```

Reply:
That makes indeed much more sense. The final goal is to remove the `Image` object and `ImageCollection` classes from the `CultuurNet\UDB3\Media\Image` namespace and only work with the new classes from the `CultuurNet\UDB3\Model\ValueObject\MediaObject\` namespace. 

But that requires even more work ... to be continued ... 😏